src/game/Spell.cpp

```diff
@@ -5100,9 +5100,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                     else
                     {
                         Pet* dbPet = new Pet;
-                        if (dbPet->LoadPetFromDB((Player*)m_caster, 0))
-                            return SPELL_CAST_OK;           // still returns an error to the player, so this error must come from somewhere else...
-                        else
+                        if (!dbPet->TryLoadFromDB((Player*)m_caster, 0))
```

Comment:
@cyberium told me back when I suggested doing the check to see if the pet is dead in SummonEffect that it should be done in CheckCast and we shouldn't return any cast errors inside the spell effects themselves.

https://github.com/cmangos/mangos-tbc/commit/2271ecbeffa490f61b6ea4cc5910011ecd738c39